File: lib/src/embedded/boundings/scaleBar/application/codec.dart
part of mapbox_navigation_flutter;

class ScaleBarAPICodec extends StandardMessageCodec {
  const ScaleBarAPICodec() : super();

  @override
  void writeValue(WriteBuffer buffer, Object? value) {
    if (value is ScaleBarSettings) {
      buffer.putUint8(128);
      writeValue(buffer, value.encode());
    } else {
      super.writeValue(buffer, value);
    }
  }

  @override
  Object? readValueOfType(int type, ReadBuffer buffer) {
    if (type == 128) {
      return ScaleBarSettings.decode(readValue(buffer)!);
    } else {
      return super.readValueOfType(type, buffer);
    }
  }
}

File: android/src/main/kotlin/com/eopeter/fluttermapboxnavigation/boundings/scaleBar/application/Codec.kt
package com.eopeter.fluttermapboxnavigation.boundings.scaleBar.application

import com.eopeter.fluttermapboxnavigation.core.*
import io.flutter.plugin.common.StandardMessageCodec
import java.io.ByteArrayOutputStream
import java.nio.ByteBuffer

@Suppress("UNCHECKED_CAST")
object ScaleBarApiCodec : StandardMessageCodec() {
    override fun readValueOfType(type: Byte, buffer: ByteBuffer): Any? {
        return when (type) {
            128.toByte() -> {
                return (readValue(buffer) as? List<Any?>)?.let {
                    ScaleBarSettings.fromList(it)
                }
            }
            else -> super.readValueOfType(type, buffer)
        }
    }
    override fun writeValue(stream: ByteArrayOutputStream, value: Any?) {
        when (value) {
            is ScaleBarSettings -> {
                stream.write(128)
                writeValue(stream, value.toList())
            }
            else -> super.writeValue(stream, value)
        }
    }
}